spec/models/concerns/digital_object_concerns/validations/dynamic_fields_validator_spec.rb

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe DigitalObject::DynamicFieldsValidator do
+  let(:project) { FactoryBot.create(:project) }
+  let(:item) { FactoryBot.create(:item, primary_project: project) }
+
+  # Setting up descriptive_metadata fields
+  let(:field_definitions) do
+    {
+      dynamic_field_categories: [
+        {
+          display_label: 'Sample Dynamic Field Category',
+          dynamic_field_groups: [
+            {
+              string_key: 'group1',
+              display_label: 'Group 1',
+              dynamic_fields: [
+                { string_key: 'string_field', display_label: 'String Field', field_type: DynamicField::Type::STRING },
+                { string_key: 'integer_field', display_label: 'Integer Field', field_type: DynamicField::Type::INTEGER }
+              ]
+            }
+          ]
+        }
+      ]
+    }
+  end
+
+  before do
+    Hyacinth::DynamicFieldsLoader.load_fields!(field_definitions, load_vocabularies: true)
+    item.assign_descriptive_metadata({ 'descriptive_metadata' => descriptive_metadata }, false)
+  end
+
+  context 'when new value is being added to a descriptive field' do
```

Comment:
Since the only thing inside this context is a single context (no `let`, `before`, second `context`, etc.), I'd remove the outer context and modify the inner message as needed.